### PR TITLE
arrays_strings_hometask

### PR DESCRIPTION
**Pull request by TMakarevich**
Hi,
Some tasks were solved in a workaround way (it seems to me), for sure there is a more correct option, but I need more time to learn different methods)

The last task in strings.js I tried to perform without regular expressions but did not get an effective result. The decision is attached:
[example.txt](https://github.com/TMakarevich/js-mentoring-tmakarevich/files/4883104/example.txt)

### CODE REVIEW

File: tasks/arrays/arrays.js [hunk 1]
```diff
@@ -4,13 +4,17 @@
...
 }
 
 /**
  * print (console.log) ids of all characters
  * @param {Array} chars
  */
 function printCharacterNames(chars) {
+    chars.forEach(char => {
+        console.log(char.name);
+    });
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js]: one-line solution:
chars.forEach(char => console.log(char.name));

File: tasks/arrays/arrays.js [hunk 2]
```diff
@@ -35,6 +43,8 @@ function getJerryInfo(chars) {
  * @return {boolean}
  */
 function isAllHuman(chars) {
+    let allHuman = char => char.species == 'Human';
+    return (chars.every(allHuman));
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js]: return  chars.every(char => char.species == 'Human');

File: tasks/arrays/arrays.js [hunk 3]
```diff
@@ -43,6 +53,8 @@ function isAllHuman(chars) {
  * @return {boolean}
  */
 function isAnyFishPerson(chars) {
+    let AnyFishPerson = char => char.type == 'Fish-Person';
+    return (chars.some(AnyFishPerson));
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js]: return  chars.some(char => char.type == 'Fish-Person');

File: tasks/strings/strings.js [hunk 4]
```diff
@@ -4,23 +4,38 @@
  * @param {string} str
  */
 function reverseString(str) {
+  if (typeof (str) == 'number') return 'This is not a string!';
```
Comment (dshaplyko) [inline tasks/strings/strings.js]: it is better to check in a more wide-spread way: type of str !== 'string'

File: tasks/strings/strings.js [hunk 5]
```diff
@@ -4,23 +4,38 @@
...
+  let arr = str.split('');
+  let arrReverse = arr.reverse();
+  let strReverse = arrReverse.join('');
+  return strReverse;
```
Comment (dshaplyko) [inline tasks/strings/strings.js]: one-line:
return str.split('').reverse().join('');

File: tasks/strings/strings.js [hunk 6]
```diff
@@ -4,23 +4,38 @@
...
 /**
  * 1. Calculate a century by given year
  * if a number is passed
-* @param {number} year
-*/
+ * @param {number} year
+ */
 function centuryFromYear(year) {
-
+  let str = String(year); // the logic for non 4 digits year is not realised
```
Comment (dshaplyko) [inline tasks/strings/strings.js]: it is toooooo complex :)
please use Math.ceil() here

File: tasks/strings/strings.js [hunk 7]
```diff
@@ -4,23 +4,38 @@
...
 }
 
 /**
  * Calculate count of the provided char in the string
  * @param {string} str
  * @param {string} char
  */
-function strCount(str, char) {
+function strCount(str, char) {
+  let count = 0;
```
Comment (dshaplyko) [inline tasks/strings/strings.js]: another solution:
let regex = new RegExp( char, 'g' );

  if (str.match(regex) == null) {
    return 0;
  } else {
    return str.match(regex).length;
  }

File: tasks/strings/strings.js [hunk 8]
```diff
@@ -40,7 +58,14 @@ function truncateString(str, num) {
...
 function replace10(text) {
-
+  let arr = text.split('');
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] == 1 && arr[i + 1] == 0) {
+      arr.splice(i, 2, 'ten');
+    }
+  }
+  let textNew = arr.join('');
+  return textNew;
```
Comment (dshaplyko) [inline tasks/strings/strings.js]: function replace10(text) {
  let re = /10/gi;
  let  newstr = text.replace(re, 'ten');
  return newstr;
}

**Review by dshaplyko — CHANGES_REQUESTED** [review]
Thank you for your work! Please consider my comments and fix some of the items

File: tasks/arrays/arrays.js (hunk 1 above)
Comment (TMakarevich) [reply tasks/arrays/arrays.js]: fixed

File: tasks/arrays/arrays.js (hunk 2 above)
Comment (TMakarevich) [reply tasks/arrays/arrays.js]: fixed

File: tasks/arrays/arrays.js (hunk 3 above)
Comment (TMakarevich) [reply tasks/arrays/arrays.js]: fixed

File: tasks/strings/strings.js (hunk 4 above)
Comment (TMakarevich) [reply tasks/strings/strings.js]: fixed

File: tasks/strings/strings.js (hunk 5 above)
Comment (TMakarevich) [reply tasks/strings/strings.js]: fixed,
but it is very difficult to perceive yet these long constructions)

File: tasks/strings/strings.js (hunk 6 above)
Comment (TMakarevich) [reply tasks/strings/strings.js]: fixed

File: tasks/strings/strings.js (hunk 8 above)
Comment (TMakarevich) [reply tasks/strings/strings.js]: fixed

File: tasks/strings/strings.js (hunk 7 above)
Comment (TMakarevich) [reply tasks/strings/strings.js]: fixed,
but requires clarification)

**Review by dshaplyko — APPROVED** [review]
Approved, please merge it)

**Pull request merged**